### PR TITLE
Autocorrect email

### PR DESCRIPTION
**Pull request by darkodosenovic**
:elephant: :art:

Implementing solution to suggest correction of domain for emails without forcing user to change it.

### CODE REVIEW

File: app/assets/javascripts/ama_layout/autocorrect_email.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+var $email = $('[type="email"]');
+var $hint = $("*#email_hint");
+
+$email.on('blur' , function() {
+  $.each( $email, function( index, value ) {
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/autocorrect_email.js]: Extra spaces inside `( )`

File: app/assets/javascripts/ama_layout/autocorrect_email.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    $.each( $hint, function( hint_index, hint_value) {
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/autocorrect_email.js]: Extra spaces inside `( )`

File: app/assets/javascripts/ama_layout/autocorrect_email.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          $($hint[hint_index]).html(suggestion).fadeIn(150);
+        }
+      });
+    })
+
+  });
+});
+
+$hint.on('click', '.email_domain', function() {
+  $.each( $email, function( index, value ){
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/autocorrect_email.js]: Extra spaces inside `( )`

File: app/assets/javascripts/ama_layout/autocorrect_email.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+var $email = $('[type="email"]');
+var $hint = $("*#email_hint");
+
+$email.on('blur' , function() {
+  $.each( $email, function( index, value ) {
+    $.each( $hint, function( hint_index, hint_value) {
+      $(value).mailcheck({
+        suggested: function(element, suggestion) {
+          var suggestion = "Did you mean <span class='suggestion'>" +
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/autocorrect_email.js]: Could you use a different variable name, so it doesn't match the parameter?

File: app/assets/javascripts/ama_layout/autocorrect_email.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            "<span class='address'>" + suggestion.address + "</span>" +
+            "@<a href='#' class='email_domain'>" + suggestion.domain +
+            "</a></span>?";
+
+          $($hint[hint_index]).html(suggestion).fadeIn(150);
+        }
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/autocorrect_email.js]: Maybe add the empty function as per the mailcheck doc?

```
,
    empty: function(element) {
      $($hint[hint_index]).html("");
    }
```

**Comment by MathieuGilbert** [comment]
:book:

File: app/assets/javascripts/ama_layout/autocorrect_email.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+var $email = $('[type="email"]');
+var $hint = $("*#email_hint");
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/autocorrect_email.js]: Do you know what the markup for email fields and email_hints will look like?

File: app/assets/stylesheets/ama_layout/layout_components/forms.scss [hunk 7]
```diff
@@ -189,6 +189,19 @@ small.error{
   margin-top: $base-margin*8;
 }
 
+#email_hint{
+  color: $brand-red;
+  display: none;
+  font-size: 0.875em;
+  margin: -10px 0 10px;
```
Comment (kathleenAMA) [inline app/assets/stylesheets/ama_layout/layout_components/forms.scss]: You can use $base-padding*2 instead of static pixels

**Comment by darkodosenovic** [comment]
:wave:

**Comment by MathieuGilbert** [comment]
I can't do the review now since I had my fingers all over that JS code.

File: app/assets/javascripts/ama_layout/desktop/index.js [hunk 8]
```diff
@@ -8,3 +8,4 @@
 //= require ./sticky-footer
 //= require ./foundation-namespace
 //= require ./sidebar
+//= require ./mailcheck
```
Comment (mvandenbeuken) [inline app/assets/javascripts/ama_layout/desktop/index.js]: newline

**Comment by darkodosenovic** [comment]
:wave:

**Comment by mvandenbeuken** [comment]
:elephant: :elephant: :elephant:

**Comment by kathleenAMA** [comment]
:art: :art: :art:

**Pull request merged**